src/Stryker.Core/Stryker.Core/Reporters/JsonReporter.cs

```diff
@@ -1,82 +1,263 @@
-﻿using Newtonsoft.Json;
+﻿using Microsoft.CodeAnalysis;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
 using Stryker.Core.Initialisation.ProjectComponent;
 using Stryker.Core.Mutants;
 using Stryker.Core.Options;
 using System.Collections.Generic;
 using System.IO;
+using System.IO.Abstractions;
 using System.Linq;
 
 namespace Stryker.Core.Reporters
 {
     public class JsonReporter : IReporter
     {
         private readonly StrykerOptions _options;
-        public JsonReporter(StrykerOptions options)
+        private readonly IFileSystem _fileSystem;
+        public JsonReporter(StrykerOptions options, IFileSystem fileSystem)
```

Comment:
In other classes we do `IFileSystem fileSystem = null` and
` _fileSystem = fileSystem ?? new FileSystem();`